internal/appSetup.go

```diff
@@ -0,0 +1,139 @@
+package internal
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/jacobsa/fuse"
+	"github.com/jacobsa/fuse/fuseutil"
+
+	"os/signal"
+
+	"sync"
+	"syscall"
+	"time"
+
+	"github.com/urfave/cli"
+	"golang.org/x/net/context"
+
+	daemon "github.com/sevlyar/go-daemon"
+)
+
+var waitedForSignal os.Signal
+
+func waitForSignal(wg *sync.WaitGroup) {
+	signalChan := make(chan os.Signal, 1)
+	signal.Notify(signalChan, syscall.SIGUSR1, syscall.SIGUSR2)
+
+	wg.Add(1)
+	go func() {
+		waitedForSignal = <-signalChan
+		wg.Done()
+	}()
+}
+
+func kill(pid int, s os.Signal) (err error) {
+	p, err := os.FindProcess(pid)
+	if err != nil {
+		return err
+	}
+
+	defer p.Release()
+
+	err = p.Signal(s)
+	if err != nil {
+		return err
+	}
+	return
+}
+
+func Mount(ctx context.Context, mountPoint string, gen3FuseConfig *Gen3FuseConfig, manifestURL string) (fs *Gen3Fuse, mfs *fuse.MountedFileSystem, err error) {
+	fs, err = NewGen3Fuse(ctx, gen3FuseConfig, manifestURL)
+	if fs == nil {
+		err = fmt.Errorf("Mount: initialization failed")
+		return
+	}
+	server := fuseutil.NewFileSystemServer(fs)
+
+	// Mount the file system.
+	mountCfg := &fuse.MountConfig{
+		FSName:                  "gen3fuse",
+		ErrorLogger:             nil,
+		DisableWritebackCaching: true,
+	}
+
+	mfs, err = fuse.Mount(mountPoint, server, mountCfg)
+	if err != nil {
+		err = fmt.Errorf("Mount: %v", err)
+		return
+	}
+
+	return
+}
+
+func Unmount(mountPoint string) (err error) {
+	err = fuse.Unmount(mountPoint)
+	return err
+}
+
+func InitializeApp(gen3FuseConfig *Gen3FuseConfig, manifestURL string, mountPoint string) {
+	app := cli.NewApp()
+	var child *os.Process
+
+	app.Action = func(c *cli.Context) (err error) {
+		defer func() {
+			time.Sleep(time.Second)
+		}()
+
+		var wg sync.WaitGroup
+		waitForSignal(&wg)
+
+		daemonCtx := new(daemon.Context)
+		child, err = daemonCtx.Reborn()
```

Comment:
Why are you doing this daemonize dance and passing signals around?

Reply:
This is adapted from Goofys code. I believe it runs FUSE as a background process, which I find a lot more convenient.